### PR TITLE
feat: add a long-running operation

### PR DESCRIPTION
**Pull request by odsod**
Make DeleteShipper a long-running operation to enable POC:ing with how
to best support authorization for the implicit LRO mixins.

### CODE REVIEW

File: proto/src/einride/iam/example/v1/freight_service.proto [hunk 1]
```diff
@@ -346,6 +351,17 @@ service FreightService {
     };
   }
 
+  option (einride.iam.v1.operations_permissions) = {
+    operation_permissions {
```
Comment (alfredgunnar) [inline proto/src/einride/iam/example/v1/freight_service.proto:355]: Mismatch in naming, I think. We should probably either change to `einride.iam.v1.operation_permissions` or to `operations_permissions { ... }`.

File: proto/src/einride/iam/example/v1/freight_service.proto [hunk 2]
```diff
@@ -346,6 +351,17 @@ service FreightService {
...
+      operation {type: "iam-example.einride.tech/ShipperOperation"}
```
Comment (alfredgunnar) [inline proto/src/einride/iam/example/v1/freight_service.proto:356]: I always thought this was some kind of JSON syntax, but this sure doesn't look like it. I guess it's some special syntax then?

Comment (odsod) [reply proto/src/einride/iam/example/v1/freight_service.proto:356]: Yes, it's actually proto text syntax: https://pkg.go.dev/google.golang.org/protobuf/encoding/prototext

**Review by alfredgunnar — APPROVED** [review]
LGTM! 👍🏼

File: proto/src/einride/iam/example/v1/freight_service.proto (hunk 1 above)
Comment (odsod) [reply proto/src/einride/iam/example/v1/freight_service.proto]: I updated naming to make the top-level more explicitly named `longrunning_operations_permissions` - the second level is permissions for a single operation - that's why operation is singular here!

Comment (alfredgunnar) [reply proto/src/einride/iam/example/v1/freight_service.proto]: Perfect!

**Pull request merged**